### PR TITLE
[python-package] fix sklearn n_jobs/nthreads and seed/random_state bug 

### PR DESCRIPTION
**Pull request by wxchan**
discuss in https://github.com/dmlc/xgboost/commit/6cea1e3fb7bdc435aeba5ea6fc5567e29249486e#commitcomment-22391065

if n_jobs/nthreads or seed/random_state is changed in other place except `__init__`, it will cause clone error.

### CODE REVIEW

**Comment by wxchan** [comment]
@ClimbsRocks check this pr, I think it will fix the error you mention.

**Comment by wxchan** [comment]
@gaw89 I didn't find `warnings.simplefilter('default', DeprecationWarning)`. Do you mean changing Line 16 https://github.com/wxchan/xgboost/blob/36797008735be7b199adee8e3ff95bc160f558b8/python-package/xgboost/sklearn.py#L16? `warnings.simplefilter('always', DeprecationWarning)` to `warnings.filterwarnings('default', '.+nthread.+|.+seed.+', DeprecationWarning)`?

**Comment by gaw89** [comment]
@wxchan, sorry for being unclear.  That is what I was meaning.  Line 16.  Perhaps this is bad practice to slip in a fix for multiple bugs under one PR, but it seemed like this might be a convenient place to fix it (rather than creating a whole PR for just one line).  If it's a bad idea to make that fix here, I'm happy to submit it in a separate PR.

**Comment by ClimbsRocks** [comment]
Thanks for the prompt attention!

Thanks to @mglowacki100 's comment https://github.com/ClimbsRocks/auto_ml/issues/232#issuecomment-306024671, i've got a folder that reproduces the error. 
[aml_test.zip](https://github.com/dmlc/xgboost/files/1052753/aml_test.zip)

I didn't include XGBoost in there, and you'll have to `pip install auto_ml`, but otherwise it should be set. 

Unfortunately, I tried checking out this commit and then rebuilding XGBoost, and it didn't seem to stop the error. 

I found something interesting as well. I made a change to `auto_ml` master, where I removed all references to `nthread`, and still got the `nthread` deprecation warning. 

I haven't tried building anything other than the master branch of xgb before, so i might have messed that part up, but hopefully this folder helps you reproduce the error!

**Comment by wxchan** [comment]
sorry I am outside these days.

@ClimbsRocks I cannot reproduce error with your provided codes in aml_test.zip. seem already solved.

@gaw89 I suggest opening another pr, because I am not quite familiar with python warnings, afraid to get something wrong.

btw, who should I call for a review for this thread?
